perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -41,6 +56,109 @@ public void setPerunBl(PerunBl perunBl) {
 		this.perunBl = perunBl;
 	}
 
+
+	@Override
+	public Consent createConsent(PerunSession sess, Consent consent) throws ConsentExistsException, UserNotExistsException, PrivilegeException, ConsentHubNotExistsException, ConsentNotExistsException {
+		Utils.notNull(consent, "consent");
+
+		// Check if UNSIGNED consent exists and delete it
+		try {
+			Consent oldConsent =  getConsentForUserAndConsentHub(sess, consent.getUserId(), consent.getConsentHub().getId(), ConsentStatus.UNSIGNED);
+			if (oldConsent != null) {
+				deleteConsent(sess, oldConsent);
+			}
+		} catch (ConsentNotExistsException e) {
+			// OK
+		}
+
+		// Add attributes from ConsentHub
+		ConsentHub consentHub = getConsentHubById(sess, consent.getConsentHub().getId());
+		Set<AttributeDefinition> filter = new HashSet<>(consent.getAttributes());
```

Comment:
```suggestion
		Set<AttributeDefinition> filter = new HashSet<>();
```
Just a detail, so the original consent attributes are ignored, as the Javadoc says.